### DISCUSSION TITLE
How can I pass my setup file as a parameter?

### DISCUSSION THREAD

**Discussion by karypid** [eclipse-oomph/oomph#108] [General]
I've defined my own product as in file "myclipse.setup". When I start the eclipse installer, I switch to advanced mode, press the + icon to add to the product list, browse the file-system to select that file and am then able to install it.

My question is: how can I run the installer and tell it to just use that setup file to install the latest version of my product?

I've only found [this discussion](https://github.com/orgs/eclipse-oomph/discussions/24) which links to this project https://github.com/a-langer/eclipse-oomph-console but I can't get that to work. Is there no easy way to simply pass a setup file to the standard installer?

**Comment by karypid** [comment]
I managed to get this to work, but in addition to the actual product.setup file, I needed to create a "productcatalog.setup" and "index.setup" file.  It seems the installer requires an index file to operate.

**Reply by merks** [reply]
This is likely useful as well:

https://wiki.eclipse.org/Eclipse_Oomph_Authoring#Automation_and_Specialization_with_Configurations

I guess you looked at this already:

https://wiki.eclipse.org/Eclipse_Oomph_Authoring#Contributing_to_a_Project_Catalog

I wouldn't generally expect you'd need to your own index:

https://wiki.eclipse.org/Eclipse_Oomph_Authoring#Hosting_your_own_index_.2F_catalogs
